src/Content/Version.php

```diff
@@ -109,77 +152,70 @@ public function model(): ModelWithContent
 	/**
 	 * Returns the modification timestamp of a version
 	 * if it exists
-	 *
-	 * @param string $lang Code `'default'` in a single-lang installation
 	 */
 	public function modified(
-		string $language = 'default'
+		Language|string $language = 'default'
 	): int|null {
-		$this->ensure($language);
-		return $this->model->storage()->modified($this->id, $language);
+		if ($this->exists($language) === true) {
+			return $this->model->storage()->modified($this->id, $this->language($language));
+		}
+
+		return null;
 	}
 
 	/**
 	 * Moves the version to a new language and/or version
+	 *
+	 * @throws \Kirby\Exception\NotFoundException If the version does not exist
 	 */
-	public function move(string $fromLanguage, VersionId $toVersionId, string $toLanguage): void
-	{
+	public function move(
+		Language|string $fromLanguage,
+		VersionId $toVersionId,
+		Language|string $toLanguage
```

Comment:
`Idea`: The `toVersionId` and `toLanguage` arguments could be nullable to make them optional. So one could e.g. do:

```php
$version->move('de', toLanguage: 'en');
$version->move('de', toVersionId: VersionId::published());
```

The argument that is left out would default to its "current" value. What we would need though is an exception if neither the target version nor target language are passed. So I'm not sure if the added code is worth the usage sugar.

Reply:
We hardly ever use `::move` so far in our code. I would not optimize this to be honest, or rather optimize it later if we ever find a part where we need to work with it more often and find it annoying.